doc/languages-frameworks/neovim.section.md

```diff
@@ -0,0 +1,107 @@
+# Neovim {#neovim}
+
+Install `neovim-unwrapped` to get a barebone neovim to configure imperatively.
+Neovim can be configured to include your favorite plugins and additional libraries by installing `neovim` instead.
+See the next section for more details.
+
+## Custom configuration {#neovim-custom-configuration}
+
+For Neovim the `configure` argument can be overridden to achieve the same:
+
+```nix
+neovim.override {
+  configure = {
+    customRC = ''
+      # here your custom configuration goes!
+    '';
+  };
+}
+```
+
+If you want to use `neovim-qt` as a graphical editor, you can configure it by overriding Neovim in an overlay
+or passing it an overridden Neovim:
+
+```nix
+neovim-qt.override {
+  neovim = neovim.override {
+    configure = {
+      customRC = ''
+        # your custom configuration
+      '';
+    };
+  };
+}
+```
+
+### Specificities for some plugins {#neovim-plugin-specificities}
+#### Treesitter {#neovim-plugin-treesitter}
+
+By default `nvim-treesitter` encourages you to download, compile and install
+the required Treesitter grammars at run time with `:TSInstall`. This works
+poorly on NixOS.  Instead, to install the `nvim-treesitter` plugins with a set
+of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
+
+```nix
+(pkgs.neovim.override {
+  configure = {
+    packages.myPlugins = with pkgs.vimPlugins; {
+      start = [
+        (nvim-treesitter.withPlugins (
+          plugins: with plugins; [
+            nix
+            python
+          ]
+        ))
+      ];
+    };
+  };
+})
+```
+
+To enable all grammars packaged in nixpkgs, use `pkgs.vimPlugins.nvim-treesitter.withAllGrammars`.
+
+
+### Testing Neovim plugins {#testing-neovim-plugins}
+
+#### neovimRequireCheck {#testing-neovim-plugins-neovim-require-check}
+`neovimRequireCheck` is a simple test which checks if Neovim can requires lua modules without errors. This is often enough to catch missing dependencies.
```

Comment:
```suggestion
`neovimRequireCheck` is a simple test which checks if Neovim can load a plugin's lua modules without errors. This is often enough to catch missing dependencies.
```

Reply:
I prefer the "require" mention that makes it clearer for the initiated (which this subsection is for).